lib/kaguya/core.ex

```diff
@@ -49,6 +49,11 @@ defmodule Kaguya.Core do
     {:noreply, state}
   end
 
+  def handle_info(:rejoin_chan, state) do
```

Comment:
Won't this require users to manually trigger the rejoin? It seems better that it would happen automatically, for instance in `handle_reconnect` after sending :init to self.